tests/unit/utils/test_utils.py

```diff
@@ -644,6 +644,16 @@ def test_sanitize_filename_empty_replacement():
     assert utils.sanitize_filename(name, replacement=None) == 'Bad File'
 
 
+@hypothesis.given(filename=hypothesis.strategies.text(min_size=100),
+                  suffix=hypothesis.strategies.text())
```

Comment:
If you change the handling according to my proposal above, it might be useful to generate the root and the extension separately here?